File: leetcode/array/leetcode_605.java
package leetcode.array;

public class leetcode_605 {
    public static void main(String[] args) {
        leetcode_605 lc = new leetcode_605();
        int[] flowerbed = {1, 0, 0, 0, 1};
        System.out.println(lc.canPlaceFlowers(flowerbed, 1));
    }

    public boolean canPlaceFlowers(int[] flowerbed, int n) {
        if (n == 0) return true;

        for (int i = 0; i < flowerbed.length && n > 0; i++) {
            if (flowerbed[i] == 0 && (i == 0 || flowerbed[i - 1] == 0) && (i == flowerbed.length - 1 || flowerbed[i + 1] == 0)) {
                flowerbed[i] = 1;
                n--;
            }
        }

        return n == 0;
    }
}

File: array/LeetCode_605.kt
package array

class LeetCode_605 {

    fun canPlaceFlowers(flowerbed: IntArray, n: Int): Boolean {
        var i = 0
        var n = n

        while (i in flowerbed.indices && n > 0) {
            if (i + 1 < flowerbed.size && flowerbed[i + 1] == 1){
                i++
            } else if (flowerbed[i] != 1) {
                n--
            }
            i += 2
        }

        return n == 0
    }
}

fun main() {
    // Input
    val flowerbed = intArrayOf(1, 0, 0, 0, 1)
    val n = 1
    // Output
    LeetCode_605().canPlaceFlowers(flowerbed, n).let {
        println(it)
        require(it)
    }
}
